wordpress-mobile/WordPress-iOS
# Show blogging reminders setup flow after publishing for the first time

Fixes #16745

This PR allows to show the blogging reminders setup flow after publishing a post or a story post for the first time

To test:
Prerequisite: it is recommended to have an account with at least three sites.
Test on iPhone and iPad
- Build/run, and select a site that has never set blogging reminders before
- Create a new post and publish it
- verify that after the editor has been dismissed, the blogging reminders setup flow starts
- Go to another site that had never set blogging reminders (or clear `UserDefaults` or delete and reinstall)
- Create a new story post and publish it
- verify that after the story editor has been dismissed, the blogging reminders setup flow starts
- Go to another site that had never set blogging reminders (or clear `UserDefaults` or delete and reinstall)
- Go to blog posts, drafts
- Create and save a post as draft if you don't already have one
- on that post, tap more, then publish now then publish (alternatively, open the post and then tap on the three dots and Publish then Publish Now)
- Verify that  the blogging reminders setup flow starts after the post has published
- In all the above cases, verify that the publishing notice appears after you dismiss any of the blogging reminders setup screen

## Regression Notes
1. Potential unintended areas of impact
This PR has some changes to the post publishing actions, but mainly adds optional closures that present the blogging reminders flow, so there should not be any impact on the existing functionalities.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
smoke test the publishing flow

3. What automated tests I added (or what prevented me from doing so)
none
PR submission checklist:
